Handle null values returned by Path#getFileName()

cghislai opened the pull request.

cghislai committed:
Handle null values returned by Path#getFileName()

ymajoros commented:
- I'd like to know in which circumstances that can be empty, and what we should do about it.
- That means that 2 empty names would be considered the same :-/

I'll look into it.

cghislai commented:
Addressed your concerns.
Extracted a method to return the non-null path name string for logging.
Handle the null case more appropriately in the other cases
